backend/hitas/models/utils.py

```diff
@@ -49,3 +50,59 @@ def validate_building_id(value: Optional[str]) -> None:
             params={"value": value},
         )
     # TODO: verify building id with the check digit
+
+
+# fmt: off
+_SSN_CHECK_DIGITS = [
+    "0", "1", "2", "3",
+    "4", "5", "6", "7",
+    "8", "9", "A", "B",
+    "C", "D", "E", "F",
+    "H", "J", "K", "L",
+    "M", "N", "P", "R",
+    "S", "T", "U", "V",
+    "W", "X", "Y",
+]
+# fmt: on
+
+
+def validate_social_security_number(value: str) -> bool:
+    if value is None:
+        return False
+
+    ssn_match = re.search(r"^(\d{6})([A-FYXWVU+-])(\d{3})([\dA-Z])$", value.upper())
+    if ssn_match is None:
+        return False
+
+    date_str = ssn_match.group(1)
+    century_sign = ssn_match.group(2)
+    individual_number_str = ssn_match.group(3)
+    check_digit = ssn_match.group(4)
+
+    # Figure out the century from the century sign
+    match century_sign:
+        case "A" | "B" | "C" | "D" | "E" | "F":
+            century = 2000
+        case "-" | "Y" | "X" | "W" | "V" | "U":
```

Comment:
TIL the century sign can be other than `A` or `-` 😄

Reply:
not yet, but those are valid starting from 2023 :) so this is future-proofed! :grin: